spec/The-OpenFF-System.md

```diff
@@ -0,0 +1,185 @@
+
+### Batch modification of parameters
+
+Some amount (all?) of parameters will be exposed to modification by the user.
+
+Say, for example, one wanted to scale all partial charges in a system by a constant factor.
+
+```python3
+factor = 0.8
+electrostatics = system.get_handler('Electrostatics')
+for potential in electrostatics.potentials:
+    potential.charge *= factor
+```
+
+Each handler will need to define its own rules for what modifications are allowed and, after modification, what status check(s) must pass.
+
+### User control over system combination
+
+A plug-in architecture will expose settings for the level of rigour executed in the internal consistency checks. The default will be strict, resulting in errors if there is any ambiguity in the physical description of each system (i.e. different non-bonded cutoff treatments). A small number (1-2) of other settings with more granularity will be exposed, i.e. one that is problematively permissive and another that implements some amount of "reasonable" discrepancies to be fudged together. This will all be constructed in a way that enables users to define their own sets of "knobs" for when system combination can and cannot be allowed, i.e. if aromaticity models need not match but cut-off treatments must.
+
+## Usage examples
+
+### Creating a system from scratch
+
+### Piecing together a system from scattered components
+
+### Adding systems together
+
+### Partial modification and re-parametrization
+
+## Implementation details
+
+### Pydantic BaseModel
+
+The core components are built off of the `BaseModel` class of [Pydantic](https://pydantic-docs.helpmanual.io/), which provides a framework for rigorous type validation and serialization pathways through dictionaries.
+
+### Python API
+
+The reference implementation is the [openff-system](https://github.com/openforcefield/openff-system) Python package, based off of the MolSSI [CMS cookiecutter](https://github.com/molssi/cookiecutter-cms). It includes a Pythonic API, automated unit and integration tests, versioning, and continuous integration, and extensive linting of the source code. It is developed and maintained by the Open Force Field Initiative and released under the MIT license.
+
+### Representation of physical quantities
+
+Pint. Most of the time an "out of the box" implementation, but some cases will require special treatment.
+
+## Relevant edge cases
+
+* Conception of "pre-" and "post-typing" topologies
+* Do virtual sites go in the topology, a special "post-typing" topology, or should they be computed on-the-fly as needed?
```

Comment:
My instinct is part of the topology, it defines the geometry, is affected by parameterization.